grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/multiqueue/MultiQueueTaskManager.java

```diff
@@ -74,6 +77,7 @@ public MultiQueueTaskManager(EngineID engineId) {
 
         this.producer = ConfigHelper.kafkaProducer();
         EngineCacheProvider.init(EngineCacheDistributed.init(zookeeper));
+        LockProvider.init(new ZookeeperLock(zookeeper, LOCK));
```

Comment:
I hadn't thought of this. At the moment the LockProvider can only house one lock- so while you can have multiple locks you will only be able to access one. We will definitely need to change this.